servers/visual/visual_server_scene.cpp

```diff
@@ -2641,8 +2641,19 @@ void VisualServerScene::_prepare_scene(const Transform p_cam_transform, const Ca
 	RID *directional_light_ptr = &light_instance_cull_result[light_cull_count];
 	directional_light_count = 0;
 
-	// directional lights
-	{
+	if (bool(GLOBAL_GET("rendering/quality/shadows/update_every_2_frames"))) {
+		// Toggle between directional and point light shadow updating every frame.
+		// This update staggering avoids stuttering by splitting CPU/GPU load across frames.
+		if (shadow_map_update == ShadowMapUpdate::SHADOW_MAP_UPDATE_DIRECTIONAL) {
+			shadow_map_update = ShadowMapUpdate::SHADOW_MAP_UPDATE_POINT;
+		} else {
+			shadow_map_update = ShadowMapUpdate::SHADOW_MAP_UPDATE_DIRECTIONAL;
+		}
+	} else {
+		shadow_map_update = ShadowMapUpdate::SHADOW_MAP_UPDATE_POINT_AND_DIRECTIONAL;
+	}
+
```

Comment:
If this is just to rotate between 3 modes, we can alternatively just cast to an int and increment then wraparound to zero.